### PR TITLE
[EGD-8148] Download asset documentation update

### PR DESCRIPTION
**Pull request by pholat**
EGD-8148

related documentation updated, written. Obsolete documentation removed

### CODE REVIEW

File: doc/CI.md [hunk 1]
```diff
@@ -1,32 +1,29 @@
...
 
-## future improvements
+## what we have
 
-- [] Common ccache for all builds
-- [] Add Clang-tidy and persistent reports
-- [] Daily builds moved to CI
-- [] Code Coverage for based on Daily builds
+* all steps are written down in `Jenkinsfile`, these steps use our build library on Jenkins
+* we have:
```
Comment (SP2FET) [inline doc/CI.md:19]:
```suggestion
* we already have:
```

File: doc/CI.md [hunk 2]
```diff
@@ -1,32 +1,29 @@
...
+    * daily builds on CI from which daily releases artifacts are created
+    * release builds on CI from which release artifacts are created
+    * ccached builds
```
Comment (SP2FET) [inline doc/CI.md:22]:
```suggestion
    * ccache'd builds
```

File: doc/CI.md [hunk 3]
```diff
@@ -1,32 +1,29 @@
...
+* drop draft & multiple requests for builds are dropped ( to not choke the builds)
```
Comment (SP2FET) [inline doc/CI.md:23]:
```suggestion
* drafts & multiple requests for builds are dropped ( to not choke the build's queue)
```

File: doc/build_targets.md [hunk 4]
```diff
@@ -19,15 +18,63 @@ Some build targets are enabled/disabled based on the selected target architectur
...
+|RT1051| \<Product>\-UpdatePackage      | PurePhone-\<version\>-RT1051-Update.tar           | Creates Update package, that can be used by Mudita Center or an update script|
+|linux | check                          |                                                   | build and run unittests |
+|common| assets                         |                                                   | Mudita company private assets |
+|common| json-images-target             |                                                   | Mudita company public assets |
+|RT1051| ecoboot.bin                    |                                                   | OS bootloader |
+|RT1051| updater.bin                    |                                                   | OS updater |
+
+## Product binary target
+
+Just software binary be loaded either by:
```
Comment (SP2FET) [inline doc/build_targets.md:38]:
```suggestion
The software binary can be loaded by:
```

File: doc/build_targets.md [hunk 5]
```diff
@@ -19,15 +18,63 @@ Some build targets are enabled/disabled based on the selected target architectur
...
+- JLink
+- Directly onto the device by:
+    - Updater utility
+    - direct memory access
+
+Devices in field have keys burned, therefore additional step - signing them will be required. Without signature devices will show warning on boot.
```
Comment (SP2FET) [inline doc/build_targets.md:44]:
```suggestion
Devices in the field have keys burned, therefore an additional step - binary signing will be required. Without the signature, devices will show a warning on boot.
```

File: doc/build_targets.md [hunk 6]
```diff
@@ -19,15 +18,63 @@ Some build targets are enabled/disabled based on the selected target architectur
...
+
+## Disk image
+
+Image of software to be loaded onto the device memory. Has all neccessary data to properly run firmware.
```
Comment (SP2FET) [inline doc/build_targets.md:48]:
```suggestion
Image of the software to be loaded onto the device memory. Has all necessary data to properly run the OS. 
```

File: doc/build_targets.md [hunk 7]
```diff
@@ -19,15 +18,63 @@ Some build targets are enabled/disabled based on the selected target architectur
...
+- Directly onto the device by:
+    - Updater utility
+    - direct memory access
+
+Devices in field have keys burned, therefore additional step - signing them will be required. Without signature devices will show warning on boot.
+
+## Disk image
+
+Image of software to be loaded onto the device memory. Has all neccessary data to properly run firmware. 
+Disk image can be build with addtional, development assets. Please look for `WITH_DEVELOPMENT_FEATURES` for more information in make.
```
Comment (SP2FET) [inline doc/build_targets.md:49]:
```suggestion
The disk image can be built with additional, development assets. Please look for `WITH_DEVELOPMENT_FEATURES` for more information in make.
```

File: doc/build_targets.md [hunk 8]
```diff
@@ -19,15 +18,63 @@ Some build targets are enabled/disabled based on the selected target architectur
...
+
+## Update package
+
+Tar-file meant to update software with the updater utility. It has all necessary update assets and metadata.
```
Comment (SP2FET) [inline doc/build_targets.md:53]:
```suggestion
Tar-file meant to update software with the updater utility. It has all the necessary update assets and metadata.
```

File: doc/build_targets.md [hunk 9]
```diff
@@ -19,15 +18,63 @@ Some build targets are enabled/disabled based on the selected target architectur
...
+Currently update package has all base data that should be on the device, though to make updates smaller minimal updates packages, as well gzipped packages should be easily possible.
```
Comment (SP2FET) [inline doc/build_targets.md:54]:
```suggestion
Currently, the update package has all of the base data that should be on the device, though to make updates smaller, minimal updates packages, as well as gzipped packages should be easily possible.
```

File: doc/build_targets.md [hunk 10]
```diff
@@ -19,15 +18,63 @@ Some build targets are enabled/disabled based on the selected target architectur
...
+
+## Check
+
+Builds and executes all unittests in the repository for selected product.
```
Comment (SP2FET) [inline doc/build_targets.md:58]:
```suggestion
Builds and executes all of the unit tests in the repository for the selected product.
```

File: doc/build_targets.md [hunk 11]
```diff
@@ -19,15 +18,63 @@ Some build targets are enabled/disabled based on the selected target architectur
...
+Uses [download_assets](./download_assets.md) unittests required assets (fonts)
```
Comment (SP2FET) [inline doc/build_targets.md:59]:
```suggestion
Uses [download_assets](./download_assets.md) to download unit test's required assets (fonts)
```

File: doc/build_targets.md [hunk 12]
```diff
@@ -19,15 +18,63 @@ Some build targets are enabled/disabled based on the selected target architectur
...
+
+## json-images-target
+
+Downloads Mudita public assets with [download_assets](./download_assets.md).
+
+## ecoboot.bin
+
+Downloads the bootloader with [download_assets](./download_assets.md).
+
+OS bootloader, used to pre-init hardware and launch either OS or updater utility, uses
```
Comment (SP2FET) [inline doc/build_targets.md:74]: uses what? :P

File: doc/build_targets.md [hunk 13]
```diff
@@ -19,15 +18,63 @@ Some build targets are enabled/disabled based on the selected target architectur
...
+
+## updater.bin
+
+Downloads the updater with [download_assets](./download_assets.md).
 
+Updater is used to perform firmware update via update packages.
```
Comment (SP2FET) [inline doc/build_targets.md:80]:
```suggestion
Updater is used to perform the firmware upgrade via update packages. 
```

File: doc/download_assets.md [hunk 14]
```diff
@@ -52,7 +53,7 @@ git config user.apitoken
 git config user.apitoken
 ```
 
-**if not**  please add token ( see doc: [documentation](../doc/download_asset.md)
+**if not**  please add token ( see doc: [documentation](./doc/download_assets.md#github-api-token)
```
Comment (SP2FET) [inline doc/download_assets.md:56]:
```suggestion
**if not**  please add the token ( see doc: [documentation](./doc/download_assets.md#github-api-token)
```

File: tools/download_asset.py [hunk 15]
```diff
@@ -143,9 +161,21 @@ class FileCache:
     file cache utility, right now - file cache is in cache dir
     if any improvement was to be made we should just take any ready
     library (i.e. DiskCache)
+
+    caching mechanism very simple - it uses cache directory where elements are stored after unpacking after download
```
Comment (SP2FET) [inline tools/download_asset.py:165]:
```suggestion
    caching mechanism is very simple - it uses a cache directory where elements are stored after unpacking after download
```

File: tools/download_asset.py [hunk 16]
```diff
@@ -267,6 +318,17 @@ class GitOps(BaseOps):
     '''
     Simplest github download wrapper based on ghapi (passed by api)
     please see reference here: https://ghapi.fast.ai/fullapi.html
+
+    It can either:
+        - load release file and unpack files one by one
+        - load files straight from the repository
+
+    *WARNING* please mind that current limit for filesize to download via API from the repository (not release)
+    if limited to 1MB via API.
```
Comment (SP2FET) [inline tools/download_asset.py:327]:
```suggestion
    *WARNING* please mind that the current limit for filesize to download via API from the repository (not the release)
    if limited to 1MB (server-side).
```

**Review by wojtekrzepecki — APPROVED** [review]
small things

**Review by SP2FET — APPROVED** [review]

File: doc/build_targets.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (pholat) [reply doc/build_targets.md]: outhdated :)

**Pull request merged**